### PR TITLE
Add check for tweets that have been edited

### LINKED ISSUE #194: Add option to override "skip legacy verified users" for legacy blues who actually use blue features
I'm generally ok with Blue-Blocker leaving legacy checkmarks alone. However, if someone is a legacy checkmark, but then actually goes and uses the new Blue features, that to me represents a problem. Specifically, if I see the "See More" that comes from the higher character limit, or formatted text, I'd want to override my "skip legacy verified users" setting and have Blue-Blocker mute them for me anyways.

### PR DESCRIPTION
**Pull request by rougetimelord**
## Changelog
- [FEAT] Check tweet results for signs of previous editing, addresses some of #195 
- [FEAT] Add option to block legacy verified users if they use blue features, close #194 

## Deployment Checklist
2. [x] ensure `src/manifest.ts` and `package.json` have the correct version number
3. [ ] use makefile to generate zips (`make chrome`, `make firefox`)
    - [x] chrome should be tested with `npm run build`
    - [ ] test firefox locally using zip
4. [ ] merge llb to main
5. [ ] upload zips from `3` to chrome webstore and firefox addons

### CODE REVIEW

**Comment by cooljeanius** [comment]
Does this do anything for #194?

File: src/parsers/instructions.ts
Comment (cooljeanius) [inline src/parsers/instructions.ts:42]: hm, this is all it takes? Interesting...

**Comment by rougetimelord** [comment]
> Does this do anything for #194?

I don't think so? It just sets user.is_blue_verified to true but then the following check can override that.  https://github.com/kheina-com/Blue-Blocker/blob/2c1e8ea5ea0ea2b7df9d03a6095c49ba15c0d6ff/src/shared.ts#L556-L572

We could add a property to the user object if we detect blue features being used and skip the legacy check if it's true to close #194.

File: src/parsers/instructions.ts
Comment (rougetimelord) [reply src/parsers/instructions.ts:42]: I believe so, if a tweet hasn't been edited that property doesn't exist. So I think `ptr?.edit_control?.edit_tweet_ids?.initial_tweet_id ` would evaluate to undefined.

There's also a list of tweet ids at `ptr.edit_control.edit_tweet_ids` that seems like it's always present, and will only have a length greater than 1 if the tweet was edited.

**Comment by rougetimelord** [comment]
Ok, made it so that the (non)usage of blue features that the extension detects gets reflected in the user object and the legacy DB check is skipped if usage is detected. Probably should add a config option to control that behavior but tbh I really don't like editing the popup code 😭, @cooljeanius if you want to add that it would be much appreciated ❤️

File: src/parsers/instructions.ts
Comment (cooljeanius) [reply src/parsers/instructions.ts:42]: ok, cool

**Review by cooljeanius — APPROVED** [review]
LGTM; haven't tested it myself, but I assume you have?

**Comment by rougetimelord** [comment]
> LGTM; haven't tested it myself, but I assume you have?

Gunna test it for a bit longer, and try to get the option added to config before merging :)

**Comment by rougetimelord** [comment]
Putting this PR on hold while we see how the update removing checkmark hiding plays out.

**Comment by rougetimelord** [comment]
No more functionality changes, good to go for reviews. 

Tested, and looks good on Chrome. I think it's worth merging now, since there is a use for the functionality suggested in #194 even if users can't hide checkmarks anymore.

**Review by cooljeanius — APPROVED** [review]
LGTM; I haven't tested it myself, though

**Pull request merged**
